### ISSUE TITLE
in Gitpod - docker-compose MTU value should be set to 1440

### ISSUE THREAD

**Issue by shaal** [drud/ddev#3766]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Since Gitpod removed slirp4netns as part of performance improvements, MTU value should be aligned to the one in gitpod.io

Gitpod fixed it for docker - https://github.com/gitpod-io/gitpod/pull/9356 and for docker-compose - https://github.com/gitpod-io/template-docker-compose/pull/4

ddev doesn't use Gitpod's custom docker-compose binary. Instead, ddev uses its own docker-compose binary at /home/gitpod/.ddev/bin/docker-compose

I tried setting ddev's global config `use_docker_compose_from_path: true`, but it resulted in this error
```
Error response from daemon: create .: volume name is too short, names should be at least two alphanumeric characters 
Failed to start drupalpod: ComposeCmd failed to run 'COMPOSE_PROJECT_NAME=ddev-drupalpod docker-compose -f /workspace/DrupalPod/.ddev/.ddev-docker-compose-full.yaml up --build -d', action='[up --build -d]', err='exit status 1', stdout='#1 [drud/ddev-dbserver-mariadb-10.3:v1.19.0-drupalpod-built internal] load build definition from Dockerfile
```

### Expected Behavior

We should find a solution for ddev projects that run in Gitpod.

The current working solution is adding docker-compose yaml file with the correct setting https://github.com/shaal/DrupalPod/pull/84/files

Possible better solutions (so we don't have to implement it in ever ddev-in-gitpod project:
* ddev uses Gitpod's docker-compose binary when run in a Gitpod workspace
* ddev sets MTU to 1440 when running in Gitpod

### Anything else?

Additional context -
https://github.com/gitpod-io/gitpod/issues/9354
https://github.com/gitpod-io/gitpod/pull/9356
https://github.com/shaal/DrupalPod/pull/84

**Comment by rfay** [comment]
I understand that this is just all about solving a problem some way, but it's not right to be worrying about MTU at the application level. This is a gitpod bug.

**Comment by chrisfromredfin** [comment]
I should confirm that Ofer and I couldn't replicate the issue on our local
DDEVs, only on Gitpod, from inside Docker. However, I do understand that
the MTU inside the containers needs to be lower than it is on the host, and
that for DDEV I think I was seeing MTU's of 65536 (presumably the max?)
locally. But again, to reiterate, we couldn't replicate in local
DDEV instances (only Gitpod).

--
Christopher J. Wells
Redfin Solutions, LLC
office: 908-4-DRUPAL
fax: 207-292-1414
***@***.***
www.redfinsolutions.com

**Comment by rfay** [comment]
Gitpod's custom docker-compose is v2.4.1.2-gitpod - there are various fatal bugs in docker-compose v2.4+, so not chasing those, but I assume there's an unresolvable problem with 2.4.1 that prevents you from using it with `ddev config global --use-docker-compose-from-path`

**Comment by rfay** [comment]
The common internet MTU is 1500...  Where does the 1440 come from? And... why wouldn't the normal value already be in play?

It seems that the fixes gitpod has done are all in the wrong place - I think they should be setting the MTU on the docker0 interface (which is still set to 1500).

**Comment by rfay** [comment]
Opened https://github.com/drud/ddev/pull/3773 - if you can please give an authoritative test and execute it yourself it will be much appreciated.

**Comment by shaal** [comment]
@rfay I followed the instructions of the video you shared with me - https://homenetworkgeek.com/mtu-size/
TLDR: find the largest number with no packet lost, and add `28` to that number.

In Gitpod, `ddev ssh`
`ping www.drupal.org -s 1412` was successful, any number above it had packet lost.
(1412 + 28 = 1440)

By the way, if you ping `www.homenetworkgeek.com` instead of Drupal, you can reach a higher number, I think that's related somehow to the fact that d.o uses Fastly as its CDN.

**Comment by rfay** [comment]
I guess we can get the interface MTU value in the hardcoded way they do it, https://github.com/gitpod-io/gitpod/pull/9356/files#diff-39f21ac00b904269148785bf38d0d67aa3bd012ef024c43c4be9480ebdbe38edR126-R132, but of course it's dependent on them keeping the name the same over time (ceth0). Not sure that hard-wiring the MTU isn't better.
